copy/copy.go

```diff
@@ -249,9 +250,18 @@ func Image(ctx context.Context, policyContext *signature.PolicyContext, destRef,
 		return nil, fmt.Errorf("determining manifest MIME type for %s: %w", transports.ImageName(srcRef), err)
 	}
 
+	// Prepare copySingleImageOptions
+	var compressionFormat *compression.Algorithm
+	var compressionLevel *int
+	if c.options.DestinationCtx != nil {
+		compressionFormat = c.options.DestinationCtx.CompressionFormat
+		compressionLevel = c.options.DestinationCtx.CompressionLevel
+	}
```

Comment:
This is now an unnecessary duplication of code centralized in `copySingleImage`. Same for the `multiple.go` copy.

Reply:
Yeah I don't need to set these I can just set in `instanceCopyClone` case.